ampersand-state.js

```diff
@@ -386,8 +386,18 @@ _.extend(Base.prototype, BBEvents, {
                 options = options || {};
 
                 var newVal = def.fn.call(self);
+                var newValToCompare, cachedValToCompare;
 
-                if (self._cache[name] !== newVal || !def.cache) {
+                if (newVal.toJSON && self._cache[name].toJSON) {
```

Comment:
Maybe add a comment here to explain what's happening? I know the source doesn't have many comments, but no reason to continue the bad habit right? ;)